### PR TITLE
Refactor Some Code

### PR DESCRIPTION
**Pull request by BenMorganIO**
Hey guys,

I was going through the open source code and reasoning about the implementation. I noticed there was some code that could have been refactored. I hope this helps :)

### CODE REVIEW

File: lib/ex_machina.ex [hunk 1]
```diff
@@ -141,21 +141,24 @@ defmodule ExMachina do
...
 
-  defp do_merge(%{__struct__: _} = record, attrs) do
-    struct!(record, attrs)
-  end
-  defp do_merge(record, attrs) do
-    Map.merge(record, attrs)
+  defp build_function_name(factory_name) do
+    factory_name
+    |> Atom.to_string
+    <> "_factory"
```
Comment (paulcsmith) [inline lib/ex_machina.ex]: Could you do `Kernel.<>("_factory")` here instead?

File: lib/ex_machina.ex [hunk 2]
```diff
@@ -141,21 +141,24 @@ defmodule ExMachina do
...
+    |> String.to_atom
   end
 
+  defp do_merge(%{__struct__: _} = record, attrs), do: struct!(record, attrs)
+  defp do_merge(record, attrs), do: Map.merge(record, attrs)
```
Comment (paulcsmith) [inline lib/ex_machina.ex]: Nice!

File: lib/ex_machina.ex [hunk 3]
```diff
@@ -177,20 +180,16 @@ defmodule ExMachina do
       build_list(3, :user)
   """
   def build_list(module, number_of_factories, factory_name, attrs \\ %{}) do
-    Enum.map(1..number_of_factories, fn(_) ->
+    Enum.map 1..number_of_factories, fn(_) ->
```
Comment (paulcsmith) [inline lib/ex_machina.ex]: Yeah that looks nicer 👍

File: lib/ex_machina.ex [hunk 4]
```diff
@@ -177,20 +180,16 @@ defmodule ExMachina do
...
-    end)
+    end
   end
 
   defmacro __before_compile__(_env) do
     quote do
-      @doc """
-      Raises a helpful error if no factory is defined.
-      """
+      @doc "Raises a helpful error if no factory is defined."
```
Comment (paulcsmith) [inline lib/ex_machina.ex]: Good catch

File: lib/ex_machina/ecto.ex [hunk 5]
```diff
@@ -230,7 +226,7 @@ defmodule ExMachina.Ecto do
   defp insert_built_belongs_to_assoc(module, association, record) do
     case Map.get(record, association.field) do
       built_relation = %{__meta__: %{state: :built}} ->
-        relation = built_relation |> module.insert
+        relation = module.insert(built_relation)
```
Comment (paulcsmith) [inline lib/ex_machina/ecto.ex]: Good idea. That reads better

File: lib/ex_machina/ecto.ex [hunk 6]
```diff
@@ -263,10 +258,8 @@ defmodule ExMachina.Ecto do
...
 
   defp convert_atom_keys_to_strings(struct) do
-    Enum.reduce(
-      struct,
-      Map.new,
-      fn({key, value}, acc) -> Map.put(acc, to_string(key), value) end
-    )
+    Enum.reduce struct, Map.new, fn({key, value}, acc) ->
+      Map.put(acc, to_string(key), value)
+    end
```
Comment (paulcsmith) [inline lib/ex_machina/ecto.ex]: Much better

File: lib/ex_machina/sequence.ex [hunk 7]
```diff
@@ -22,15 +20,11 @@ defmodule ExMachina.Sequence do
        ExMachina.Sequence.reset
      end
   """
-  def reset do
-    Agent.update(__MODULE__, fn(_) -> Map.new end)
-  end
+  def reset, do: Agent.update(__MODULE__, fn(_) -> Map.new end)
```
Comment (paulcsmith) [inline lib/ex_machina/sequence.ex]: For this and `start_link` I think I prefer the previous version. This is a bit too busy IMO

File: lib/ex_machina.ex (hunk 1 above)
Comment (paulcsmith) [inline lib/ex_machina.ex]: Sorry for the confusion. I meant to type `|> Kernel.<>("_factory")` that way you can keep the pipe chain going

**Comment by BenMorganIO** [comment]
@paulcsmith I've reverted and rebased the changes with `start_link` and `reset`. I've also added 81ffde1 for some areas that I missed.

File: lib/ex_machina.ex (hunk 1 above)
Comment (BenMorganIO) [inline lib/ex_machina.ex]: Oh, good catch. Running this code in my `iex` and seeing that it doesn't actually work...

Comment (BenMorganIO) [inline lib/ex_machina.ex]: And fixed :p

Comment (paulcsmith) [inline lib/ex_machina.ex]: 👍 :D

**Comment by paulcsmith** [comment]
Thanks a ton @BenMorganIO!

File: lib/ex_machina.ex (hunk 1 above)
Comment (pdawczak) [inline lib/ex_machina.ex]: Sorry for jumping to the party and possibly silly question, but why not simply:

```elixir
"#{factory_name}_factory"
```

Comment (BenMorganIO) [inline lib/ex_machina.ex]: @pdawczak that's a very very good question :p

**Pull request merged**